sympy/solvers/diophantine.py

```diff
@@ -381,45 +392,196 @@ def diop_linear(eq, param=symbols("t", integer=True)):
 
 
 def _diop_linear(var, coeff, param):
```

Comment:
Why is this a private function? I think the function is useful enough to become the part of the public api.